public/app/app.ts

```diff
@@ -74,54 +74,59 @@ export class GrafanaApp {
-    addClassIfNoOverlayScrollbar();
-    setLocale(config.bootData.user.locale);
-    setWeekStart(config.bootData.user.weekStart);
-    setPanelRenderer(PanelRenderer);
-    setTimeZoneResolver(() => config.bootData.user.timezone);
-    // Important that extensions are initialized before store
-    initExtensions();
-    configureStore();
-
-    standardEditorsRegistry.setInit(getAllOptionEditors);
-    standardFieldConfigEditorRegistry.setInit(getStandardFieldConfigs);
-    standardTransformersRegistry.setInit(getStandardTransformers);
-    variableAdapters.setInit(getDefaultVariableAdapters);
-    monacoLanguageRegistry.setInit(getDefaultMonacoLanguages);
-
-    setQueryRunnerFactory(() => new QueryRunner());
-    setVariableQueryRunner(new VariableQueryRunner());
-
-    locationUtil.initialize({
-      config,
-      getTimeRangeForUrl: getTimeSrv().timeRangeForUrl,
-      getVariablesUrlParams: getVariablesUrlParams,
-    });
-
-    // intercept anchor clicks and forward it to custom history instead of relying on browser's history
-    document.addEventListener('click', interceptLinkClicks);
-
-    // disable tool tip animation
-    $.fn.tooltip.defaults.animation = false;
-
-    this.angularApp.init();
-
-    // Preload selected app plugins
-    const promises = [];
-    for (const modulePath of config.pluginsToPreload) {
-      promises.push(importPluginModule(modulePath));
-    }
+  async init() {
+    try {
+      initEchoSrv();
+      addClassIfNoOverlayScrollbar();
+      setLocale(config.bootData.user.locale);
+      setWeekStart(config.bootData.user.weekStart);
+      setPanelRenderer(PanelRenderer);
+      setTimeZoneResolver(() => config.bootData.user.timezone);
+      // Important that extensions are initialized before store
+      initExtensions();
+      configureStore();
+
+      standardEditorsRegistry.setInit(getAllOptionEditors);
+      standardFieldConfigEditorRegistry.setInit(getStandardFieldConfigs);
+      standardTransformersRegistry.setInit(getStandardTransformers);
+      variableAdapters.setInit(getDefaultVariableAdapters);
+      monacoLanguageRegistry.setInit(getDefaultMonacoLanguages);
+
+      setQueryRunnerFactory(() => new QueryRunner());
+      setVariableQueryRunner(new VariableQueryRunner());
+
+      locationUtil.initialize({
+        config,
+        getTimeRangeForUrl: getTimeSrv().timeRangeForUrl,
+        getVariablesUrlParams: getVariablesUrlParams,
+      });
+
+      // intercept anchor clicks and forward it to custom history instead of relying on browser's history
+      document.addEventListener('click', interceptLinkClicks);
+
+      // disable tool tip animation
+      $.fn.tooltip.defaults.animation = false;
+
+      this.angularApp.init();
+
+      // Preload selected app plugins
+      const promises: Array<Promise<any>> = [];
+      for (const modulePath of config.pluginsToPreload) {
+        promises.push(importPluginModule(modulePath));
+      }
+
+      await Promise.all(promises);
```

Comment:
:thinking: I did not realize preload happened before the grafana app initalizes.

I don't think a failed plugin load should ever cause grafana to not load.

Can we load "preload" plugins after the app initializes and catch each plugin error explicitly?  Normally we have an error boundary around each plugin

Reply:
yea, I said this also when we last made a change to this, this should happen after app boot and should not trigger the app fail load view

Reply:
Ok ok, I will open a separate PR to address this.